src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/InstructionSetSupport.cs

```diff
@@ -0,0 +1,350 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+using ILCompiler.DependencyAnalysis.ReadyToRun;
+using Internal.TypeSystem;
+
+namespace ILCompiler
+{
+    public class InstructionSetSupport : ICompilationRootProvider
```

Comment:
Once this is in Common, it will trigger the rule that says "stuff in Common shouldn't reference the ReadyToRun namespace".

Instead of implementing `ICompilationRootProvider`, could you move the piece of logic in that method to the `Compilation` constructor where we can just `_dependencyGraph.AddRoot` it? It will also avoid having to expose `NodeFactory` on `IRootingServiceProvider`.